pkg/services/sqlstore/migrations/accesscontrol/team_membership.go

```diff
@@ -361,24 +187,26 @@ func (p *teamPermissionMigrator) sortRolesToAssign(userPermissionsByOrg map[int6
 				return nil, nil, nil, errFindingRoles
 			}
 
-			roleKey := p.getAssignmentKey(orgID, roleName)
+			if rolesByOrg[orgID] == nil {
+				rolesByOrg[orgID] = map[string]*accesscontrol.Role{}
+			}
 
 			if role.ID != 0 {
-				rolesByOrg[roleKey] = &role
+				rolesByOrg[orgID][roleName] = &role
 			} else {
 				roleToCreate := &accesscontrol.Role{
 					Name:  roleName,
 					OrgID: orgID,
 				}
 				rolesToCreate = append(rolesToCreate, roleToCreate)
 
-				userAssignments, initialized := assignments[userID]
+				userAssignments, initialized := assignments[orgID]
```

Comment:
nit: these aren't really `userAssignments` anymore, seems like `orgAssignments` might be more suitable.

Reply:
Had the same thought, which lead to https://github.com/grafana/grafana/pull/44702#discussion_r816131165